clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1677,7 +1677,9 @@ void Sema::MarkDevice(void) {
         case attr::Kind::SYCLIntelMaxGlobalWorkDim:
         case attr::Kind::SYCLIntelMaxWorkGroupSize:
         case attr::Kind::SYCLIntelNoGlobalWorkOffset: {
-          SYCLKernel->addAttr(A);
+          Attr *NewA = A->clone(getASTContext());
```

Comment:
There are two other locations where this happens (addAttr), the same needs to be done there as well, at least from a correctness perspective.